pkg/iterator/emitters.go

```diff
@@ -76,7 +81,112 @@ func (e *TableEmitter) SnapshotMetadataIteratorRecord(recordNumber int, metadata
 		fmt.Fprintf(e.Writer, tableRowFmt, recordNumber, metadata.VolumeCapacityBytes, bmt, bmd.ByteOffset, bmd.SizeBytes)
 	}
 
-	return true
+	return nil
+}
+
+func (e *TableEmitter) SnapshotMetadataIteratorDone(_ int) error {
+	return nil
+}
+
+// VerifierEmitter will write the changed blocks from the source device to the target device at the designated offset and
+// compare the contents of the source and target devices at the end.
+type VerifierEmitter struct {
+	// SourceDevice contains the source device file descriptor.
+	SourceDevice *os.File
+
+	// TargetDevice contains the target device file descriptor.
+	TargetDevice *os.File
 }
 
-func (e *TableEmitter) SnapshotMetadataIteratorDone(_ int) {}
+// copyBlock will copy chunks of 256 bytes at a time upto to sizeBytes from the source device to the target device.
+func (verifierEmitter *VerifierEmitter) copyBlock(bmd *api.BlockMetadata) error {
+	// Seek to the byteOffset of the source device.
+	_, err := verifierEmitter.SourceDevice.Seek(bmd.ByteOffset, io.SeekStart)
+	if err != nil {
+		return fmt.Errorf("failed to seek source device(%q) to offset(%d): %w", verifierEmitter.SourceDevice.Name(), bmd.ByteOffset, err)
+	}
+
+	// Seek to the byteOffset of the target device.
+	_, err = verifierEmitter.TargetDevice.Seek(bmd.ByteOffset, io.SeekStart)
+	if err != nil {
+		return fmt.Errorf("failed to seek target device(%q) to offset(%d): %w", verifierEmitter.TargetDevice.Name(), bmd.ByteOffset, err)
+	}
+
+	const chunkSize = 256
```

Comment:
512 is the minimum Unix block size, and in most cases page size blocks are okay. Given that you're splitting the data stream and handling the trailing smaller chunk in your loop below, it should be safe to allocate a much larger buffer.

Reply:
> 512 is the minimum Unix block size, and in most cases page size blocks are okay. Given that you're splitting the data stream and handling the trailing smaller chunk in your loop below, it should be safe to allocate a much larger buffer.

Set it to 1024*1024 now.